ui/src/table.tsx

```diff
@@ -727,35 +728,50 @@ export const
         const expandedRef = expandedRefs[key]
         return expandedRef === undefined || expandedRef
       },
+      initialGroups = React.useMemo(() => {
+        return m.groups
+          ? m.groups.reduce((acc, { label }) => {
+            acc[label] = ({ key: label, name: label, startIndex: 0, count: 0 })
+            return acc
+          }, {} as Dict<Fluent.IGroup>)
+          : groupByKey !== '*'
+            ? items.reduce((acc, item) => {
+              const group = (item as Fluent.IObjectWithKey & Dict<any>)[groupByKey]
+              if (!acc[group]) acc[group] = ({ key: group, name: group, startIndex: 0, count: 0 })
+              return acc
+            }, {} as Dict<Fluent.IGroup>)
+            : {}
+      }, [m.groups, groupByKey, items]),
       makeGroups = React.useCallback((groupByKey: S, filteredItems: (Fluent.IObjectWithKey & Dict<any>)[]) => {
+        const allGroups: Dict<Fluent.IGroup> = { ...initialGroups }
         let
           groups: Fluent.IGroup[],
           groupedBy: Dict<any> = []
 
         if (m.groups) {
-          groups = filteredItems.reduce((acc, { group }, idx) => {
-            const prevGroup = acc[acc.length - 1]
-            prevGroup?.key === group
-              ? prevGroup.count++
-              : acc.push({ key: group, name: group, startIndex: idx, count: 1, isCollapsed: getIsCollapsed(group, expandedRefs.current) })
-            return acc
-          }, [] as Fluent.IGroup[])
+          filteredItems.forEach(({ group }, idx) => {
+            allGroups[group].count === 0
+              ? allGroups[group] = ({ key: group, name: group, startIndex: idx, count: 1, isCollapsed: getIsCollapsed(group, expandedRefs.current) })
+              : allGroups[group].count++
+          })
+          groups = Object.values(allGroups)
         } else {
           let prevSum = 0
           groupedBy = groupByF(filteredItems, groupByKey)
           const
             groupedByKeys = Object.keys(groupedBy),
             groupByColType = m.columns.find(c => c.name === groupByKey)?.data_type
 
-          groups = groupedByKeys.map((key, i) => {
+          groupedByKeys.forEach((key, i) => {
```

Comment:
Isn't there a bug? If the column is a `time` data type, all groups should have the same format, not just non-empty ones. Assuming by code only though, so you gotta try it out. If so, please fix and add accompanying unit test.

Reply:
Fixed.